### PR TITLE
Remove phony prerequisites from the build target

### PR DESCRIPTION
**Pull request by skitt**
See individual commits for details. The result of this PR is that two successive `make build` invocations won’t result in two rebuilds: the second invocation will see that it has nothing to do.

### CODE REVIEW

**Comment by skitt** [comment]
Ah, this needs to use the updated vendoring support from Shipyard.

File: Makefile [hunk 1]
```diff
@@ -166,33 +166,35 @@ bin/subctl-%: generate-embeddedyamls $(shell find pkg/subctl/ -name "*.go") vend
 
 ci: generate-embeddedyamls golangci-lint markdownlint unit build images
 
-generate-embeddedyamls: generate pkg/subctl/operator/common/embeddedyamls/yamls.go
+generate-embeddedyamls: pkg/subctl/operator/common/embeddedyamls/yamls.go
```
Comment (mkolesnik) [inline Makefile]: Can't this target be dropped then? It's only used in operator and I doubt anyone is running it manually.

Also this line is the same as 171.

File: Makefile [hunk 2]
```diff
@@ -201,12 +203,14 @@ generate-clientset: vendor/modules.txt
...
-	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt,year=$(shell date +"%Y")" paths="./..."
+# Generate deep-copy code
+CONTROLLER_DEEPCOPY := apis/submariner/v1alpha1/zz_generated.deepcopy.go
+generate: $(CONTROLLER_DEEPCOPY)
+$(CONTROLLER_DEEPCOPY): $(CONTROLLER_GEN) $(VENDOR_MODULES)
+	cd apis && $(CONTROLLER_GEN) object:headerFile="$(CURDIR)/hack/boilerplate.go.txt,year=$(shell date +"%Y")" paths="./..."
 
 # Generate manifests e.g. CRD, RBAC etc
-manifests: generate $(CONTROLLER_GEN) vendor/modules.txt
+manifests: generate $(CONTROLLER_GEN) $(VENDOR_MODULES)
```
Comment (mkolesnik) [inline Makefile]:
```suggestion
manifests: $(CONTROLLER_DEEPCOPY) $(CONTROLLER_GEN) $(VENDOR_MODULES)
```

File: Makefile [hunk 3]
```diff
@@ -166,33 +166,35 @@ bin/subctl-%: generate-embeddedyamls $(shell find pkg/subctl/ -name "*.go") vend
 
 ci: generate-embeddedyamls golangci-lint markdownlint unit build images
```
Comment (mkolesnik) [inline Makefile]:
```suggestion
ci: $(EMBEDDED_YAMLS) golangci-lint markdownlint unit build images
```

File: Makefile [hunk 4]
```diff
@@ -201,12 +203,14 @@ generate-clientset: vendor/modules.txt
 		github.com/submariner-io/submariner-operator/apis \
 		submariner:v1alpha1
 
-# Generate code
-generate: $(CONTROLLER_GEN) vendor/modules.txt
-	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt,year=$(shell date +"%Y")" paths="./..."
+# Generate deep-copy code
+CONTROLLER_DEEPCOPY := apis/submariner/v1alpha1/zz_generated.deepcopy.go
+generate: $(CONTROLLER_DEEPCOPY)
```
Comment (mkolesnik) [inline Makefile]: I reckon it's the same for this target, though I might be wrong

**Review by mkolesnik — CHANGES_REQUESTED** [review]
Thanks, some suggestions

**Review by mkolesnik — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**